src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -106,13 +110,111 @@ static TimeSeriesStorage()
+        {
+            var tree = context.Transaction.InnerTransaction.CreateTree(TimeSeriesStats);
+            using (GetTimeSeriesFullName(context, docId, name, out var slice))
+            {
+                tree.Increment(slice, count);
+            }
+        }
+
+        public (long Count, DateTime Start, DateTime End) GetStatsFor(DocumentsOperationContext context, string docId, string name)
+        {
+            var tree = context.Transaction.InnerTransaction.CreateTree(TimeSeriesStats);
+            var table = new Table(TimeSeriesSchema, context.Transaction.InnerTransaction);
+            
+            using (GetTimeSeriesFullName(context, docId, name, out var slice))
+            {
+                var count = tree.Read(slice).Reader.ReadLittleEndianInt64();
+                var start = DateTime.MinValue;
+                var end = DateTime.MaxValue;
+
+                // find the first good segment for this ts
+                foreach (var (key, value) in table.SeekByPrimaryKeyPrefix(slice, Slices.Empty, 0))
+                {
+                    var ptr = value.Reader.Read((int)TimeSeriesTable.Segment, out var size);
+                    var segment = new TimeSeriesValuesSegment(ptr, size);
+                    if (segment.NumberOfLiveEntries > 0)
+                    {
+                        var baseline = ExtractDateTimeFromKey(key);
+                        foreach (var result in segment.YieldAllValues(context, context.Allocator, baseline))
+                        {
+                            if (result.Status == TimeSeriesValuesSegment.Live)
+                            {
+                                start = result.Timestamp;
+                                break;
+                            }
+                        }
+                        break;
+                    }
+                }
+
+                // to find the last value we set to highest possible segment value (long.MaxValue) and seeking backward from there.
+                IDisposable dispose = Slice.External(context.Allocator, slice.Content.Ptr, slice.Content.Length + sizeof(long), out var lastKey);
+                *(long*)(lastKey.Content.Ptr + slice.Content.Length) = Bits.SwapBytes(long.MaxValue); 
+
+                while (table.SeekOneBackwardByPrimaryKeyPrefix(slice, lastKey, out var tvr))
+                {
+                    var segmentPtr = tvr.Read((int)TimeSeriesTable.Segment, out var size);
+                    var segment = new TimeSeriesValuesSegment(segmentPtr, size);
+
+                    var keyPtr = tvr.Read((int)TimeSeriesTable.TimeSeriesKey, out var keySize);
+                    
+                    dispose.Dispose();
+                    dispose = Slice.External(context.Allocator, keyPtr, keySize, out var keySlice);
+                    lastKey = keySlice;
+                    if (segment.NumberOfLiveEntries > 0)
```

Comment:
What if I have 100000 segments with no live values?

Reply:
It depends..
If they are deleted as a segment, we store only a single dead entry so each page will probably have several hundreds of those.. Which you still have to iterate.
If those segments are full with dead values, it is indeed a problem.

But in both cases after the cleaner did his job this shouldn't matter. 

I see 2 options:
1. We can do binary search (not sure this is a good idea.. the cost might be even higher)
2. Somehow track the first value

I don't see any nice/better way here.
I will test it on the 300M points data and see if it an issue.
If it isn't we leave this optimization for later. I think.

Reply:
Alternatively, ignore dead values and report start /end as they are.
Then let the cleaner do its job?

Reply:
Also an option

Reply:
Let's go with this

Reply:
done